routers/api/v1/api.go

```diff
@@ -398,6 +407,23 @@ func reqRepoBranchWriter(ctx *context.APIContext) {
 		ctx.Error(http.StatusForbidden, "reqRepoBranchWriter", "user should have a permission to write to this branch")
 		return
 	}
+
+	checkPublicOnly(ctx, unit.TypeCode)
+}
+
+func checkPublicOnly(ctx *context.APIContext, unitType unit.Type) {
+	if true == ctx.Data["IsApiToken"] {
+		switch unitType {
+		case unit.TypeCode:
+			publicRepo, pubRepoExists := ctx.Data["ApiTokenScopePublicRepoOnly"]
+
+			if pubRepoExists && publicRepo.(bool) &&
+				ctx.Repo.Repository != nil && ctx.Repo.Repository.IsPrivate {
+				ctx.Error(http.StatusForbidden, "reqToken", "token scope is limited to public repos")
+				return
```

Comment:
What is the difference between this and `reqToken`

Reply:
These code are outdated.